### PR TITLE
MES-4809: Implement single fault competency component

### PR DESCRIPTION
**Pull request by szabikr**
## Description

Single Fault Competency component is implemented
Test report uses Single Fault Competency component
Store is hooked up with the component

Have a list of things that's need doing, but this should go in so that team members can work with it.

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1063,7 +1063,8 @@
             "ansi-regex": {
               "version": "2.1.1",
               "bundled": true,
-              "dev": true
+              "dev": true,
```
Comment (Cullima) [inline package-lock.json:1066]: Are these changes intended?

File: src/modules/tests/test-data/cat-a-mod1/single-fault-competencies/single-fault-competencies.selector.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+): boolean => {
+  return singleFaultCompetencies[competencyName] === CompetencyOutcome.S;
+};
+
+export const hasCompetencyDangerousFault = (
+  singleFaultCompetencies: SingleFaultCompetencies,
+  competencyName: SingleFaultCompetencyNames,
+): boolean => {
+  return singleFaultCompetencies[competencyName] === CompetencyOutcome.D;
+};
```
Comment (mgoldson) [inline src/modules/tests/test-data/cat-a-mod1/single-fault-competencies/single-fault-competencies.selector.ts:28]: I would prefer to see these functions in separate helper files or a provider, particularly if they can be shared across categories, so we can avoid duplication.

**Review by mgoldson — APPROVED** [review]
LGTM!

File: src/shared/helpers/single-fault-competency.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { FaultSummary } from '../models/fault-marking.model';
+
+export const getSingleFaultCompetencyDrivingFaults = (): FaultSummary[] => {
```
Comment (Cullima) [inline src/shared/helpers/single-fault-competency.ts:3]: is this needed / used?

**Review by Cullima — APPROVED** [review]
couple of minor comments but looks good other than that.

File: src/modules/tests/test-data/cat-a-mod1/single-fault-competencies/single-fault-competencies.selector.ts (hunk 2 above)
Comment (szabikr) [reply src/modules/tests/test-data/cat-a-mod1/single-fault-competencies/single-fault-competencies.selector.ts:28]: `SingleFaultCompetencies` comes from AM1 schema. might be difficult to reuse exactly these selectors
